ts/parsers/elementparser.ts

```diff
@@ -7,49 +7,134 @@ import * as isEmpty from "lodash.isempty";
+            let slideLayoutSpNode = undefined;
+            slideLayoutSpNode = slideLayoutTables["idxTable"][placeholderIdx];
+
+            return (
+                slideLayoutSpNode !== undefined &&
+                checkPath(slideLayoutSpNode, '["p:txBody"][0]["a:p"][0]["a:pPr"][0]["$"]["lvl"]') &&
+                checkPath(slideLayoutSpNode, '["p:txBody"][0]["a:lstStyle"][0]["a:lvl1pPr"][0]["a:buNone"]') === false
+            );
+        }
+
+        return false;
+    }
+
+    public getLayoutSpNodes(slideLayoutTables, slideMasterTables) {
+        const idx =
+            getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) === undefined
+                ? undefined
+                : getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "idx"]);
+        const type =
+            getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph"]) === undefined
+                ? undefined
+                : getAttributeByPath(this.element, ["p:nvSpPr", "p:nvPr", "p:ph", "$", "type"]);
+
+        let slideLayoutSpNode = undefined;
+        let slideMasterSpNode = undefined;
+
+        if (type !== undefined) {
+            slideLayoutSpNode = slideLayoutTables["typeTable"][type];
+            slideMasterSpNode = slideMasterTables["typeTable"][type];
+            return { slideLayoutSpNode, slideMasterSpNode };
+        }
+        if (idx !== undefined) {
+            slideLayoutSpNode = slideLayoutTables["idxTable"][idx];
+            slideMasterSpNode = slideMasterSpNode["idxTable"][idx];
+            return { slideLayoutSpNode, slideMasterSpNode };
+        }
+        return { slideLayoutSpNode, slideMasterSpNode };
+    }
+
+    public getXfrmNodePosition(xfrmNode) {
+        let position: PowerpointElement["elementPosition"] = null;
+        let offset: PowerpointElement["elementOffsetPosition"] = null;
+
+        position = checkPath(xfrmNode, '["a:off"][0]["$"]') ? xfrmNode["a:off"][0]["$"] : position;
+        offset = checkPath(xfrmNode, '["a:off"][0]["$"]') ? xfrmNode["a:ext"][0]["$"] : offset;
+
+        return { position, offset };
+    }
+
+    public getPosition(slideLayoutTables, slideMasterTables) {
+        const { slideLayoutSpNode, slideMasterSpNode } = this.getLayoutSpNodes(
+            slideLayoutTables,
+            slideMasterTables
+        );
+
+        const xfrmNodePath = '["p:spPr"][0]["a:xfrm"][0]';
+        const slideXfrmNode = getValueAtPath(this.element, xfrmNodePath);
+        const slideLayoutXfrmNode = getValueAtPath(slideLayoutSpNode, xfrmNodePath);
+        const slideMasterXfrmNode = getValueAtPath(slideMasterSpNode, xfrmNodePath);
+
+        if (slideXfrmNode) {
+            return this.getXfrmNodePosition(slideXfrmNode);
+        }
+
+        if (slideLayoutXfrmNode) {
+            return this.getXfrmNodePosition(slideLayoutXfrmNode);
+        }
+
+        if (slideMasterXfrmNode) {
+            return this.getXfrmNodePosition(slideMasterXfrmNode);
+        }
```

Comment:
I guess you don't have resolve all three in the begining, you can delay resolving layout x frmNode until it is required, and then delay from master even more etc. .. slightly more efficient.

```
const slideXfrmNode = getValueAtPath(this.element, xfrmNodePath);
if (slideXfrmNode) {
       return this.getXfrmNodePosition(slideXfrmNode);
}

const slideLayoutXfrmNode = getValueAtPath(slideLayoutSpNode, xfrmNodePath);
if (slideLayoutXfrmNode) {
        return this.getXfrmNodePosition(slideLayoutXfrmNode);
}
...
```

Reply:
Good point, changed it